bounded-core/src/main/scala/io/cafienne/bounded/aggregate/AggregateRootActor.scala

```diff
@@ -45,51 +43,52 @@ trait AggregateRootActor extends PersistentActor with AggregateRootStateCreator
   /**
     * When extending the AggregateRootActor you MUST implement the handleCommand method.
     *
-    * @param command is the commands your aggregate root expects
-    * @param currentState of the Aggregate Root
-    * @return a seqency of events when everything is Right, or an Exception (Left)
-    */                                                                                //TODO replace Either (no EXCEPTION)
-  def handleCommand(command: AggregateRootCommand, currentState: AggregateRootState): Either[Exception, Seq[AggregateRootEvent]]
+    * @param command one of the commands your aggregate root expects
+    * @param state of the Aggregate Root at the moment of handling command
+    * @return a sequence of events when everything is Right, or an Failure (Left)
+    */
+  def handleCommand(command: DomainCommand, state: AggregateState): Reply
+
 
   // Below this line is the internal implementation of the Aggregate Root Actor.
-  private var internalState: Option[AggregateRootState] = None
+  private var internalState: Option[AggregateState] = None
 
-  def state: Option[AggregateRootState] = { internalState }
+  def state: Option[AggregateState] = { internalState }
 
-  private def updateState(evt: AggregateRootEvent) {
-    internalState = Some(internalState.fold(newState(evt))(state => state.update(evt)))
+  private def updateState(evt: DomainEvent) {
+    internalState = Some(internalState.fold(newState(evt))(_ update evt))
   }
 
   override def persistenceId: String = aggregateId.idAsString
 
   final def receiveCommand: Actor.Receive = {
-    case cmd: AggregateRootCommand =>
+    case cmd: DomainCommand =>
       val originalSender = sender()
-      handleCommand(cmd, internalState.fold(NoState: AggregateRootState)(state => state)) match {
-        case Right(evt) =>
-          persistAll[AggregateRootEvent](evt) { e =>
-            updateState(e)
-          }
-          originalSender ! Right(evt)
-        case Left(exc) => originalSender ! Left(CommandNotProcessedException("Could not handle command.", exc))
+      val reply = handleCommand(cmd, internalState getOrElse NoState)
+
+      reply match {
+        case Ok(events) => persistAll(events)(updateState)
+        case Ko(_) => ()
```

Comment:
What's the effect of this => () ?

Reply:
None for the execution, just my preference of documenting with code. Whenever match expression result is `Unit` and there is 'no-operation' branch I tend to leave there instance of `Unit` instead of blank space.